modules/app_component/main.tf

```diff
@@ -267,6 +267,83 @@ resource "aws_cloudwatch_log_group" "CloudWatchLogGroup" {
   }
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# Autoscaling based on CPU and MEMORY Utilisation for scaling up and down to save costs.
+# ---------------------------------------------------------------------------------------------------------------------
+# ---------------------------------------------------------------------------------------------------------------------
+# Create autoscaling target linked to ECS
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_appautoscaling_target" "ecs_target" {
+  max_capacity       = 3
+  min_capacity       = 1
+  resource_id        = "service/${var.container_runtime}/${aws_ecs_service.ecs_service.name}"
+  scalable_dimension = "ecs:service:DesiredCount"
+  service_namespace  = "ecs"
+  role_arn           = aws_iam_role.ecs-autoscale-role.arn
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# Scale up based on CPU and MEMORY Utilisation
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_appautoscaling_policy" "ecs_target_cpu" {
+  count              = var.metric_type == "CPU_UTILISATION" ? 1 : 0
+  name               = "application-scaling-policy-cpu"
+  policy_type        = "TargetTrackingScaling"
+  resource_id        = aws_appautoscaling_target.ecs_target.resource_id
+  scalable_dimension = aws_appautoscaling_target.ecs_target.scalable_dimension
+  service_namespace  = aws_appautoscaling_target.ecs_target.service_namespace
+
+  target_tracking_scaling_policy_configuration {
+    predefined_metric_specification {
+      predefined_metric_type = "ECSServiceAverageCPUUtilization"
+    }
+    target_value = 90
```

Comment:
I'd also like to configure scale-in and scale-out thresholds for cpu from terra3.

Reply:
Due to targate tracking policy, it can not be configured.